packages/core/src/odata/common/selectable/selectable.ts

```diff
@@ -22,13 +22,13 @@ export type Selectable<EntityT extends EntityBase> = ODataVersionOf<
       | Link<EntityT, any>
       | ComplexTypeField<EntityT, any>
       | CustomFieldBase<EntityT>
-      | CollectionField<EntityT>
+      | CollectionField<EntityT, any>
```

Comment:
you could define a default value for the FieldT to be any if you want to avoid changes like this. Additional question: The  `ComplexTypeField`  and `Link` have the default value:
```
export abstract class ComplexTypeField<
  EntityT extends EntityBase,
  ComplexTypeNamespaceT extends ComplexTypeNamespace = any
>
```
So I guess one could unify all the entries here to just have the Entity as a generic argument. Symmetry is good :-).

Reply:
Yeah, I don't love, that then you can use those things without an actual type, but yeah, selectable is a good example where this can make sense 😊